expression/builtin_ilike_vec.go

```diff
@@ -0,0 +1,204 @@
+	}
+
+	escape, isConstNull, err := b.args[2].EvalInt(b.ctx, chunk.Row{})
+	if isConstNull {
+		fillNullStringIntoResult(result, rowNum)
+		return escape, true, nil
+	} else if err != nil {
+		return escape, true, err
+	}
+	return escape, false, nil
+}
+
+func (b *builtinIlikeSig) lowerExpr(param *regexpParam, rowNum int) {
+	col := param.getCol()
+	if col == nil {
+		str := param.getStringVal(0)
+		str_bytes := hack.Slice(str)
+		stringutil.LowerOneString(str_bytes)
+		param.setStrVal(str)
+		return
+	}
+
+	tmpExprCol := param.getCol().CopyConstruct(nil)
+	LowerAlphaAscii(tmpExprCol, rowNum)
+	param.setCol(tmpExprCol)
+}
+
+func (b *builtinIlikeSig) lowerPattern(param *regexpParam, rowNum int, escape int64) int64 {
+	col := param.getCol()
+	if col == nil {
+		str := param.getStringVal(0)
+		str_bytes := hack.Slice(str)
+		escape = int64(stringutil.LowerOneStringExcludeEscapeChar(str_bytes, byte(escape)))
+		param.setStrVal(str)
+		return escape
+	}
+
+	tmpPatternCol := param.getCol().CopyConstruct(nil)
+	escape = LowerAlphaAsciiExcludeEscapeChar(tmpPatternCol, rowNum, escape)
+	param.setCol(tmpPatternCol)
+
+	return escape
+}
+
+func (b *builtinIlikeSig) vecVec(params []*regexpParam, rowNum int, escape int64, result *chunk.Column) error {
+	result.ResizeInt64(rowNum, false)
+	result.MergeNulls(params[0].getCol(), params[1].getCol())
+	i64s := result.Int64s()
+	for i := 0; i < rowNum; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		b.pattern.Compile(params[1].getStringVal(i), byte(escape))
+		match := b.pattern.DoMatch(params[0].getStringVal(i))
+		i64s[i] = boolToInt64(match)
+	}
+	return nil
+}
+
+func (b *builtinIlikeSig) constVec(expr string, param *regexpParam, rowNum int, escape int64, result *chunk.Column) error {
+	result.ResizeInt64(rowNum, false)
+	result.MergeNulls(param.getCol())
+	i64s := result.Int64s()
+	for i := 0; i < rowNum; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		b.pattern.Compile(param.getStringVal(i), byte(escape))
+		match := b.pattern.DoMatch(expr)
+		i64s[i] = boolToInt64(match)
+	}
+	return nil
+}
+
+func (b *builtinIlikeSig) ilikeWithMemorization(exprParam *regexpParam, rowNum int, escape int64, result *chunk.Column) error {
+	result.ResizeInt64(rowNum, false)
+	result.MergeNulls(exprParam.getCol())
+	i64s := result.Int64s()
+	for i := 0; i < rowNum; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		match := b.pattern.DoMatch(exprParam.getStringVal(i))
+		i64s[i] = boolToInt64(match)
+	}
+	return nil
+}
+
+func (b *builtinIlikeSig) ilikeWithoutMemorization(params []*regexpParam, rowNum int, escape int64, result *chunk.Column) error {
+	if params[0].getCol() == nil {
+		return b.constVec(params[0].getStringVal(0), params[1], rowNum, escape, result)
+	}
+
+	return b.vecVec(params, rowNum, escape, result)
+}
+
+func (b *builtinIlikeSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
+	rowNum := input.NumRows()
+	params := make([]*regexpParam, 0, 3)
```

Comment:
Looks like `regexpParam` is no longer a suitable name, should rename it.

Reply:
> Looks like `regexpParam` is no longer a suitable name, should rename it.

okk